### PR TITLE
reboot a VM only on managed disk resize

### LINKED ISSUE #6063: changing azurerm_managed_disk disk tags causes VM reboot
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.21
+ provider.azurerm v1.44.0

### Affected Resource(s)

* `azurerm_managed_disk`

### Terraform Configuration Files

```hcl
data "azurerm_resource_group" "rg" {
  name = "<RESOURCE_GROUP_NAME>"
}

resource "azurerm_network_interface" "vm1234-nic" {
  name                = "vm1234-nic"
  location            = data.azurerm_resource_group.rg.location
  resource_group_name = data.azurerm_resource_group.rg.name

  ip_configuration {
    name                          = "vm1234-ip"
    subnet_id = "<SUBNET_ID>"
    private_ip_address_allocation = "Dynamic"
  }
}

resource "azurerm_managed_disk" "vm1234_data_disk_1" {
  location             = data.azurerm_resource_group.rg.location
  resource_group_name  = data.azurerm_resource_group.rg.name

  name  = "vm1234_data_disk_1"
  storage_account_type = "Standard_LRS"
  disk_size_gb         = "32"
  create_option        = "Empty"
  tags = {
    "tag1" = "value1"
    #"tag2" = "value2"
  }
}

resource "azurerm_virtual_machine" "vm1234" {
  name                  = "vm1234"
  resource_group_name   = data.azurerm_resource_group.rg.name
  location = data.azurerm_resource_group.rg.location
  network_interface_ids = [azurerm_network_interface.vm1234-nic.id]
  vm_size               = "Standard_B1ls"

  storage_os_disk {
    name              = "vm1234-disk"
    caching           = "ReadWrite"
    create_option     = "FromImage"
    managed_disk_type = "Premium_LRS"
  }

  storage_image_reference {
    publisher = "Canonical"
    offer     = "UbuntuServer"
    sku       = "16.04.0-LTS"
    version   = "latest"
  }

  os_profile {
    computer_name  = "vm1234"
    admin_username = "<REDACTED>"
    admin_password = "<REDACTED>"
  }

  os_profile_linux_config {
    disable_password_authentication = false
  }

  storage_data_disk {
    name = "vm1234_data_disk_1"
    caching = "ReadWrite"
    lun = 0
    disk_size_gb = "32"
    create_option   = "Attach"
    managed_disk_id = azurerm_managed_disk.vm1234_data_disk_1.id
  }
}
```

### Debug Output

```
...
2020-03-11T00:54:31.620-0700 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: [DEBUG] Shutting Down Virtual Machine "vm1234" (Resource Group "rg")..
2020-03-11T00:54:31.620-0700 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: [DEBUG] AzureRM Request: 
2020-03-11T00:54:31.620-0700 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: POST /subscriptions/UUID/resourceGroups/rg/providers/Microsoft.Compute/virtualMachines/vm1234/powerOff?api-version=2019-07-01&skipShutdown=false HTTP/1.1
2020-03-11T00:54:31.620-0700 [DEBUG] plugin.terraform-provider-azurerm_v1.44.0_x4: Host: management.az
...
```

### Expected Behavior

changing tags on a managed disk should not trigger VM reboot

### Actual Behavior

VM is rebooted unexpectedly

### Steps to Reproduce

1. `terraform apply`
1. uncomment tag2
1. `terraform apply` watch VM rebooting, inspect VM activity log

### References

* seems to be introduced by https://github.com/terraform-providers/terraform-provider-azurerm/pull/5579

### PR DESCRIPTION
**Pull request by shurikk**
related to https://github.com/terraform-providers/terraform-provider-azurerm/issues/6063

Fixes #6063

### CODE REVIEW

File: website/docs/r/managed_disk.html.markdown [hunk 1]
```diff
@@ -101,7 +101,7 @@ The following arguments are supported:
 
 * `disk_mbps_read_write` - (Optional) The bandwidth allowed for this disk; only settable for UltraSSD disks. MBps means millions of bytes per second.
 
-* `disk_size_gb` - (Optional, Required for a new managed disk) Specifies the size of the managed disk to create in gigabytes. If `create_option` is `Copy` or `FromImage`, then the value must be equal to or greater than the source's size.
+* `disk_size_gb` - (Optional, Required for a new managed disk) Specifies the size of the managed disk to create in gigabytes. If `create_option` is `Copy` or `FromImage`, then the value must be equal to or greater than the source's size. The size can only be increased. Changing this forces a VM reboot.
```
Comment (tombuildsstuff) [inline website/docs/r/managed_disk.html.markdown]: The "this'll restart the VM" bit is covered in the yellow info box below this, so we can remove this

```suggestion
* `disk_size_gb` - (Optional, Required for a new managed disk) Specifies the size of the managed disk to create in gigabytes. If `create_option` is `Copy` or `FromImage`, then the value must be equal to or greater than the source's size. The size can only be increased.
```

**Comment by tombuildsstuff** [comment]
hey @shurikk 

Thanks for this PR - I've spent a little time digging into this and it appears there's some more edge-cases here - as such I hope you don't mind but I'm going to push a couple of commits to rebase this PR (to include the changes from #6207) and then handle those edge-cases so that we can get this merged.

Thanks!

**Comment by shurikk** [comment]
thanks 👍

**Comment by tombuildsstuff** [comment]
Ignoring one test failing due to capacity issues in Azure - the tests pass:

Screenshot 2020-03-24 at 18 01 19

**Comment by shurikk** [comment]
any chance this fix goes into 1.xx branch? I know that 1.44 was the last one in 1.xx, this bug is somewhat dangerous.
